opensearch_py_ml/ml_commons/IAMRoleHelper.py

```diff
@@ -0,0 +1,353 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import json
+import logging
+import uuid
+from datetime import datetime
+
+import boto3
+import requests
+from botocore.exceptions import ClientError
+from requests.auth import HTTPBasicAuth
+
+from opensearch_py_ml.ml_commons.cli.aws_config import AWSConfig
+from opensearch_py_ml.ml_commons.cli.opensearch_domain_config import (
+    OpenSearchDomainConfig,
+)
+
+
+class IAMRoleHelper:
+    """
+    Helper class for managing IAM roles and their interactions with OpenSearch.
+    """
+
+    NO_SUCH_ENTITY = "nosuchentity"
+    NOT_FOUND = "not_found"
+
+    def __init__(
+        self,
+        opensearch_config: OpenSearchDomainConfig,
+        aws_config: AWSConfig,
+    ):
+        """
+        Initialize the IAMRoleHelper with AWS and OpenSearch configurations.
+        """
+        self.opensearch_config = opensearch_config
+        self.aws_config = aws_config
+
+        self.iam_client = boto3.client(
+            "iam",
+            aws_access_key_id=self.aws_config.aws_access_key,
+            aws_secret_access_key=self.aws_config.aws_secret_access_key,
+            aws_session_token=self.aws_config.aws_session_token,
+        )
+        self.sts_client = boto3.client(
+            "sts",
+            region_name=self.opensearch_config.opensearch_domain_region,
+            aws_access_key_id=self.aws_config.aws_access_key,
+            aws_secret_access_key=self.aws_config.aws_secret_access_key,
+            aws_session_token=self.aws_config.aws_session_token,
+        )
+
+    def role_exists(self, role_name):
+        """
+        Check if an IAM role exists.
+        :param role_name: Name of the IAM role.
+        :return: True if the role exists, False otherwise.
+        """
+        try:
+            self.iam_client.get_role(RoleName=role_name)
+            return True
+        except ClientError as e:
```

Comment:
this is duplicated in multiple methods. Can we create a separate method to reduce code redundancy?